crates/datasources/src/bigquery/mod.rs

```diff
@@ -478,22 +478,30 @@ fn bigquery_table_to_arrow_schema(table: &Table) -> Result<ArrowSchema> {
 
     let mut arrow_fields = Vec::with_capacity(fields.len());
     for field in fields {
-        let mode = field.mode.clone();
-        let resolved_field = if let Some(mode) = mode {
-            if mode == *"REPEATED" {
-                handle_repeatable_fields(field)?
-            } else {
-                table_field_schema_to_arrow_datatype(field)?
-            }
-        } else {
-            table_field_schema_to_arrow_datatype(field)?
-        };
-        arrow_fields.push(resolved_field);
+        arrow_fields.push(table_field_schema_to_arrow_datatype(field)?);
     }
     Ok(ArrowSchema::new(arrow_fields))
 }
 
 fn table_field_schema_to_arrow_datatype(field: &BigQuerySchema) -> Result<Field> {
+    let mode = field.mode.clone();
+    if let Some(mode) = mode {
+        if mode == *"REPEATED" {
+            handle_repeatable_fields(field)
+        } else {
+            handle_nullable_fields(field)
+        }
+    } else {
+        handle_nullable_fields(field)
+    }
+}
```

Comment:
I believe this could be shortened to:
```rust
match field.mode.as_deref() {
    Some("REPEATED") => handle_repeatable_fields(field),
    _ => handle_nullable_fields(field),
}
```